client/state/sites/domains/schema.js

```diff
@@ -0,0 +1,35 @@
+export const itemsSchema = {
+	type: 'object',
+	patternProperties: {
+		'^\\d+$': {
+			type: 'object',
```

Comment:
Close, but we're missing the array. The type should be `array` with `object` items:
```javascript
type: 'array',
items: {
    type: 'object',
    properties: //the props your listed below
}
```